sql/core/src/test/scala/org/apache/spark/sql/sources/DisableUnnecessaryBucketedScanSuite.scala

```diff
@@ -241,7 +241,8 @@ abstract class DisableUnnecessaryBucketedScanSuite
 
   test("SPARK-33075: not disable bucketed table scan for cached query") {
     withTable("t1") {
-      withSQLConf(SQLConf.AUTO_BUCKETED_SCAN_ENABLED.key -> "true") {
+      withSQLConf(SQLConf.AUTO_BUCKETED_SCAN_ENABLED.key -> "true",
+          SQLConf.CAN_CHANGE_CACHED_PLAN_OUTPUT_PARTITIONING.key -> "false") {
```

Comment:
can you explain the reason a little bit?

Reply:
it's not about AQE. It affects the rule `DisableUnnecessaryBucketedScan`. If we enable `CAN_CHANGE_CACHED_PLAN_OUTPUT_PARTITIONING` then the rule `DisableUnnecessaryBucketedScan` will take effect and change output partitioning (break bucket scan)

Reply:
I think this does expose an issue. It means bucketed table scan inside table cache is always useless because of `DisableUnnecessaryBucketedScan`. Shall we update `DisableUnnecessaryBucketedScan` to take table cache into account?

Reply:
The behavior is in the range of design, see
https://github.com/apache/spark/blob/d157b2d9a71c8370612342b298f0e6757c9b9b78/sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala#L61-L70

If we do not want to affect bucket scan, we need to split this configuration with one for AQE and one for bucket scan.

Reply:
My propose is, instead of turning off `AUTO_BUCKETED_SCAN_ENABLED` for cached query, can we update `DisableUnnecessaryBucketedScan` to not disable bucket scan for queries inside the table cache stage?